### PR TITLE
Ensure Dataset has TLs, add option to disable them

### PR DESCRIPTION
**Pull request by lucabergamini**
This PR includes two components:

- Ensure we're loading a dataset with TLs;
- Allow to disable TLs from config

### CODE REVIEW

File: examples/agent_motion_prediction/agent_motion_config.yaml [hunk 1]
```diff
@@ -40,6 +40,9 @@ raster_params:
   # one of the classes we care about (cars, bikes, peds, etc.), >=1.0 filter all other agents.
   filter_agents_threshold: 0.5
 
+  # if to disable completely traffic light faces in the semantic rasterizer
```
Comment (oliver-scheel) [inline examples/agent_motion_prediction/agent_motion_config.yaml:43]: whether

File: examples/visualisation/visualisation_config.yaml [hunk 2]
```diff
@@ -40,6 +40,9 @@ raster_params:
   # one of the classes we care about (cars, bikes, peds, etc.), >=1.0 filter all other agents.
   filter_agents_threshold: 0.5
 
+  # if to disable completely traffic light faces in the semantic rasterizer
```
Comment (oliver-scheel) [inline examples/visualisation/visualisation_config.yaml:43]: see above

File: l5kit/l5kit/data/zarr_dataset.py [hunk 3]
```diff
@@ -137,11 +137,7 @@ def open(self, mode: str = "r", cached: bool = True, cache_size_bytes: int = int
         self.frames = self.root[FRAME_ARRAY_KEY]
         self.agents = self.root[AGENT_ARRAY_KEY]
         self.scenes = self.root[SCENE_ARRAY_KEY]
-        try:
-            self.tl_faces = self.root[TL_FACE_ARRAY_KEY]
-        except KeyError:
-            print(f"{TL_FACE_ARRAY_KEY} not found in {self.path}! Traffic lights will be disabled")
-            self.tl_faces = np.empty((0,), dtype=TL_FACE_DTYPE)
+        self.tl_faces = self.root[TL_FACE_ARRAY_KEY]
```
Comment (oliver-scheel) [inline l5kit/l5kit/data/zarr_dataset.py:140]: So we just crash? Wasn't the previous version better?

**Review by oliver-scheel — DISMISSED** [review]
Minors, otherwise LGTM

File: l5kit/l5kit/data/zarr_dataset.py (hunk 3 above)
Comment (lucabergamini) [reply l5kit/l5kit/data/zarr_dataset.py:140]: I agree; It was better for retro-compatibility, but the thing is that we want to completely move to traffic light datasets only. So it makes sense to hard crash here imo

File: examples/agent_motion_prediction/agent_motion_config.yaml [hunk 4]
```diff
@@ -40,6 +40,9 @@ raster_params:
   # one of the classes we care about (cars, bikes, peds, etc.), >=1.0 filter all other agents.
   filter_agents_threshold: 0.5
 
+  # whether to disable completely traffic light faces in the semantic rasterizer
```
Comment (oliver-scheel) [inline examples/agent_motion_prediction/agent_motion_config.yaml:43]: Minor: to be exact still had to be 'whether to completely disable traffic light faces in the semantic rasterizer' or 'whether to disable traffic light faces in the semantic rasterizer completely'

File: examples/visualisation/visualisation_config.yaml [hunk 5]
```diff
@@ -40,6 +40,9 @@ raster_params:
   # one of the classes we care about (cars, bikes, peds, etc.), >=1.0 filter all other agents.
   filter_agents_threshold: 0.5
 
+  # whether to disable completely traffic light faces in the semantic rasterizer
```
Comment (oliver-scheel) [inline examples/visualisation/visualisation_config.yaml:43]: see above

File: l5kit/l5kit/tests/artefacts/config.yaml [hunk 6]
```diff
@@ -36,3 +36,6 @@ raster_params:
   # e.g. 0.0 include every obstacle, 0.5 show those obstacles with >0.5 probability of being
   # one of the classes we care about (cars, bikes, peds, etc.), >=1.0 filter all other agents.
   filter_agents_threshold: 0.5
+
+  # whether to disable completely traffic light faces in the semantic rasterizer
```
Comment (oliver-scheel) [inline l5kit/l5kit/tests/artefacts/config.yaml:40]: see above

**Review by oliver-scheel — DISMISSED** [review]
LGTM

File: l5kit/l5kit/data/zarr_dataset.py (hunk 3 above)
Comment (lucabergamini) [reply l5kit/l5kit/data/zarr_dataset.py:140]: I have restored this with a stronger warning. For now we will still support loading a zarr without traffic lights but not in the rest of the workflow (e.g. you won't be able to use it in an EgoDataset). In this way we can still inspect raw legacy data if we want to

**Review by oliver-scheel — APPROVED** [review]

**Pull request merged**